edx_notifications/channels/channel.py

```diff
@@ -235,6 +235,16 @@ def bulk_dispatch_notification(self, user_ids, msg, exclude_user_ids=None, chann
         raise NotImplementedError()
 
     @abc.abstractmethod
+    def dispatch_notification_to_tag(
```

Comment:
I think we don't need to define a new method in `BaseNotificationChannelProvider` we should use existing `bulk_dispatch_notification` method to send notifications to a tag group if it is available in `channel_context`.